open-api/rest-catalog-open-api.py

```diff
@@ -1100,6 +1100,31 @@ class LoadTableResult(BaseModel):
      - `s3.session-token`: if present, this value should be used for as the session token
      - `s3.remote-signing-enabled`: if `true` remote signing should be performed as described in the `s3-signer-open-api.yaml` specification
 
+    ## GCP Configurations
+
+    The following configuration respected by the GCSFileIO
+     - `gcs.project-id`: The GCP project of the service you are running
+     - `gcs.service.host`: Optional alternative endpoint for the GCS FileIO to access (format protocol://host:port). If not set, it will use the standard Google endpoint.
+     - `gcs.decryption-key`: Option for customer-supplied AES256 key for server-side decryption of the blob
+     - `gcs.encryption-key`: Option for customer-supplied AES256 key for server-side encryption of the blob
+     - `gcs.user-project`: Option for blob's billing user project. This option is used only if the blob's bucket has requester_pays flag enabled
+     - `gcs.channel.read.chunk-size-bytes`: Optional INT that sets the minimum size that will be read by a single RPC. Read data will be locally buffered until consumed
+     - `gcs.channel.write.chunk-size-byte`: Optional INT that sets the  minimum size that will be written by a single RPC. Written data will be buffered and only flushed upon reaching this size or closing the channel.
+     - `gcs.oauth2.token`: String representation of the access token used for temporary access.
+     - `gcs.oauth2.token-expires-at`: A LONG that represents the date in Epoch milliseconds of when the token expires.
+     - `gcs.no-auth`: Boolean to explicitly configure "no authentication" for testing purposes using a GCS emulator
+     - `gcs.delete.batch-size`: Optional INT that configures the batch size used when deleting multiple files from a given GCS bucket. Defaults to 50.
```

Comment:
Should we only limit this to configuration related to authentication? This is the case of AWS as well.

Reply:
We could, but I would argue that we would still want those other configs documented in the spec somewhere for constancy sake for those working on their own FileIO implementations or Iceberg REST catalog implementations. 

If we limit what is documented in this section to auth, where do you propose we put other config that could be returned?

Reply:
Any thoughts on the above @Fokko?

Reply:
Hey @Buktoria Thanks for pinging me. 

I think the more appropriate place is to have it under the `GcsFileIO`, [similar to the `S3FileIO`](https://iceberg.apache.org/docs/nightly/aws/#s3-fileio).